### DISCUSSION TITLE
.NET Interactive Server

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/interactive#2195] [Ideas]
Is there any plan to provide a web accessible application like a docker that runs nuget package or c# codes
As a developer I can have internet address and ports and filesystem on the get go, I can run scripts on my server or run a minecraft server too with docker and I picture I can do it with .NET interactive too

**Comment by jonsequitur** [comment]
It's not on our roadmap to provide this directly but .NET Interactive can certainly be used within a web application, whether via Docker or other deployment mechanisms. For example you can reference the NuGet packages directly, create `Kernel` instances, and send them commands. (The server will also need to have the .NET SDK installed, not just the runtime.)

If you do this, you'll want to carefully restrict access to the API you expose because allowing arbitrary code to be run on your web server would generally be considered a serious security vulnerability.

**Reply by Xyncgas** [reply]
gg thank
